spine-social-button.html

```diff
@@ -0,0 +1,195 @@
+  ~ Use is subject to license terms.
+  -->
+
+<link rel="import" href="../polymer/polymer.html">
+<link rel="import" href="../paper-button/paper-button.html">
+<link rel="import" href="../iron-icon/iron-icon.html">
+<link rel="import" href="../paper-styles/paper-styles.html">
+<link rel="import" href="./spine-social-button-icons.html">
+
+
+<!--
+`spine-social-button` is a [<paper-button>](https://www.webcomponents.org/element/PolymerElements/paper-button)
+styled with an icon which represents a certain social domain (e.g. Google, Github, Facebook).
+
+Example:
+```
+  <spine-social-button provider="google">Sign up with Google</spine-social-button>
+  <spine-social-button provider="github">Sign up with Github</spine-social-button>
+  <spine-social-button provider="facebook">Sign up with Facebook</spine-social-button>
+  <spine-social-button provider="twitter">Sign up with Twitter</spine-social-button>
+
+  <spine-social-button provider="google" icon="my-icons:google">Sign in with Google</spine-social-button>
+
+  <spine-social-button icon="spine-social-button:google">Share</spine-social-button>
+  <spine-social-button icon="my-icons:twitter">Share</spine-social-button>
+```
+
+The following custom properties and mixins are available for styling:
+
+Custom property | Description | Default
+----------------|-------------|----------
+`--spine-social-button-background` | Background color of the button | specified according to `provider`
+`--spine-social-button` | Mixin applied to the <paper-button> | `{}`
+`--spine-social-button-disabled` | Mixin passed to the `--paper-button-disabled` mixin| `{}`
+`--spine-social-button-focus` | Mixin passed to the `--paper-button-raised-keyboard-focus` mixin | `{}`
+
+The default styling when `provider` property is specified:
+
+Provider property | background-color   | icon
+------------------|--------------------|------------------------------
+google            | #3367d6            | spine-social-button:google
+github            | #333               | spine-social-button:github
+facebook          | #3b5998            | spine-social-button:facebook
+twitter           | #1da1f2            | spine-social-button:twitter
+
+-->
+<dom-module id="spine-social-button">
+  <template>
+    <style>
+      :host {
+        display: inline-block;
+      }
+
+      paper-button {
+        display: flex;
+        flex-direction: row;
+        align-items: center;
+        justify-content: flex-start;
+
+        background: var(--spine-social-button-background);
+
+        @apply --spine-social-button;
+
+        --paper-button-raised-keyboard-focus: {
+          @apply --spine-social-button-focus;
+        };
+
+        --paper-button-disabled: {
+          @apply --spine-social-button-disabled;
+        };
+      }
+
+      iron-icon {
+        padding-right: 8px;
+      }
+
+      #divider {
+        display: inline-block;
+        width: 1px;
+        height: 100%;
+        background: rgba(0, 0, 0, .12);
```

Comment:
Is --dark-divider-opacity applicable here? We would get rid of one more "magical" constant this way.